src/Controller/BlogController.php

```diff
@@ -36,4 +37,25 @@ public function showArticles(ArticleRepository $articleRepository)
             'articles' => $articleRepository->findAll(),
         ]);
     }
+    #[Route('/createArticle', name: 'createArticle')]
+    public function formArticle(Request $request, EntityManagerInterface $manager): Response
+    {
+        $form = $this->createForm(ArticleCreateType::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var ArticleCreate $articleDTO */
+            $articleDTO = $form->getData();
+            $article = new Article($articleDTO->title, $articleDTO->capon, $articleDTO->content);
+            $article->setCreationDate(new DateTime());
```

Comment:
Set creation date should be the in construct. if you have a setter for creationDate it means that we can change this value every where.

```php
<?php

use DateTimeImmutable;

class Titi {
  private DateTimeImmutable $createdAt;

  public function __construct()
  {
    $this->createdAt = new DateTimeImmutable()
  }
}
```